src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -1075,6 +1078,67 @@ func (c controller) logClusterOperatorsStatus() {
 	}
 }
 
+// This is temporary function that should validate router state only in case of failure
+// It will patch router to add access logs
+// It will run http router health check to see if router is healthy on host network
+func (c controller) logRouterStatus() {
+	if !c.Status.HasError() || c.HighAvailabilityMode != models.ClusterHighAvailabilityModeNone {
+		return
+	}
+	c.log.Infof("Start checking router status")
+	var cl *models.Cluster
+	var err error
+	patched := false
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr, Timeout: dnsValidationTimeout}
+	localIp, _ := utils.GetOutboundIP()
+	if localIp == "" {
+		localIp = "localhost"
+	}
+
+	var consoleUrl string
+	// run every 30 seconds router check and log the result
+	go utils.WaitForPredicate(WaitTimeout, SummaryLogsPeriod, func() bool {
```

Comment:
Does it return true at some point?

Reply:
nope, it will exit when controller will exit